### ISSUE TITLE
Generate pke files instead of packaging them in source

### ISSUE THREAD

**Issue by sneakernets** [team-eternity/eternity#112]
This one recently stuck in my craw and it's bothering me:

We are shipping a PKE file (a renamed ZIP file) along with any source release. That's fine.

But we're also doing that in the [source code release](https://github.com/team-eternity/eternity/blob/master/base/doom/eternity.pke). When investigating Issue #111 I noticed that nowhere in the repo are the images, or any resources Eternity uses apart from the IWADS, ever accessible outright. That is a problem. Let's fix that problem.
Rationale:
- Github has the ability to view images and sounds online.
- Referencing graphics becomes easier.
- Guaranteed list of lumps that will be called in the source code.
- Easier to edit and fix new graphics without uploading massive (>256K) binaries in the git repo each time.
- Less work for contributors in the long run that want to assist with assets.

Hurdles:
- Requires a way to read in "sane" indexed image formats in the executable. We should be supporting this anyway (if we aren't already)
- Requires a script to compile the package and compress it (hey, zlib to the rescue!)
- Requires a listing of files (Should have this anyway)

**Comment by Gaerzi** [comment]
You could use [ZDoom's zipdir](https://github.com/rheit/zdoom/blob/master/tools/zipdir/zipdir.c) for that. It has a nifty -u feature where it only repacks files that were updated since the last compilation, so it makes compiling the zip a bit like compiling a program: no need to rebuild everything from zero every time.

**Comment by sneakernets** [comment]
That'll do nicely.

**Comment by Altazimuth** [comment]
We actually discussed this in the IRC channel a while back. It did seem like a good idea, though we weren't sure of the particulars of build scripts. Even without packaging now that there's directory support there would be no problem with just running the unpacked folder, which makes this idea increasingly feasible. I wholeheartedly support this, as I've ran into this issue before whilst doing inventory-branch-2016 (which changed eternity.pke at the same time as Quasar did in the master).
